tf2onnx/rewriter/custom_rnn_rewriter.py

```diff
@@ -39,63 +39,22 @@ def run(self):
         log.debug("enter custom rnn rewriter")
         return self.run_internal()
 
-    def _get_rnn_scope_name(self, while_scope_name):
-        parts = while_scope_name.split('/')
-        rnn_scope = '/'.join(parts[0:-2]) + "/"
-        log.debug("found rnn scope %s", rnn_scope)
-        return rnn_scope
-
-    def _parse_rnn_loop(self, context):
-        # check a while loop is generated by dynamic_rnn or bidirectional_rnn by
-        #
-        # 1. some patterns in _time_step in dynamic_rnn: tensor array read, tensor array write
-        # 2. some patterns in control_flow_ops.while_loop in dynamic_rnn:
-        #      cond: time < loop_bound
-        #      loop_vars: (time, output_ta, state)
-        #      time has name called "time"
-        #      iteration_cnt is added by control flow.
-
-        # be noted:
-        # 1. iteration counter does not exist in tf1.4 or earlier versions
-        # 2. if dynamic_rnn's first input is not consumed, output ta does not exist.
-        time_name = context.rnn_scope + "time"
-        ta_array_name_prefix = context.rnn_scope + "dynamic_rnn/output_"
-        iteration_counter_name = context.while_context_scope + "iteration_counter"
-
-        found_time = False
-        is_rnn_out_ta = None
-        for val in context.loop_properties.all_variables.values():
-            enter_input_node = self.g.get_node_by_output(val.enter_input_id)
-            if val.is_tensor_array:
-                ta_name = enter_input_node.get_attr("tensor_array_name").s.decode("utf-8")
-                if not ta_name.startswith(ta_array_name_prefix):
-                    is_rnn_out_ta = False
-            elif enter_input_node.name == time_name:
-                found_time = True
-                context.time_var = val
-            elif enter_input_node.name == iteration_counter_name:
-                context.iteration_var = val
-
-
-        if not found_time or is_rnn_out_ta is False:
-            log.debug("this should not be a dynamic_rnn loop, found_time: %s, is_rnn_out_ta: %s",
-                      found_time, is_rnn_out_ta)
-            return False
-
-        return True
-
     def need_rewrite(self, context):
-        context.rnn_scope = self._get_rnn_scope_name(context.while_context_scope)
+        context.rnn_scope = get_rnn_scope_name(context.while_context_scope)
 
-        if not self._parse_rnn_loop(context):
+        res = parse_rnn_loop(self.g, context.loop_properties, context.rnn_scope,
+                             context.while_context_scope)
+        if not res:
             log.debug("skip the loop due to parse_rnn_loop failed")
             return False
 
-        self._parse_time_var(context)
+        time_var, iteration_var = res
+        context.time_var = time_var
+        context.iteration_var = iteration_var
```

Comment:
seems both time_var and iteration_var are useless here, if no more reference, just remove them.

Reply:
hmm, they are used by old rewriter. remove them later when all changes about rewriter are checked in?

Reply:
I now put variables used in new rewriter into UnitRnnContext. the RnnProperty will keep intact.